### ISSUE TITLE
ResolveType in Unions

### ISSUE THREAD

**Issue by mrrbodia** [graphql-dotnet/graphql-dotnet#2174]
I used the ResolveType in my Union. Everything is fine in the documentation. I created a query using the Union subtypes in the query body. As a result, I don't get any errors, but I also don't get nodes in response.
If I use IsTypeOf in types instead of this ResolveType in Union, everything works fine. But this solution does not suit me.
So, it looks like a bug.

The example of my code:
```
    public class FailureResultType : ObjectGraphType<FailureResultSource>
    {
        public FailureResultType()
        {
            Name = "TestFailureResult";
            Description = "blablabla";

            Field<NonNullGraphType<FailureReasonEnumeration>>()
                .Name("Reason")
                .Description("The failure reason.")
                .Resolve(context => context.Source.FailureReason);
        }
    }

    public class SuccessResultType : ObjectGraphType
    {
        public SuccessResultType()
        {
            Name = "TestSuccessResult";
            Description = "blablabla";

            Field<NonNullGraphType<DictionaryGraphType>>()
                .Name("Result")
                .Description("Result of command execution.")
                .Resolve(context => context.Source);
        }
    }

    public class CheckoutStepCommandResultUnion : UnionGraphType
    {
        public CheckoutStepCommandResultUnion(
            FailureResultType failureResultType ,
            SuccessResultType successResultType)
        {
            Name = "CheckoutStepCommandResults";
            Description = "Possible results of executing command.";

            Type<FailureResultType>();
            Type<SuccessResultType>();

            ResolveType = source =>
            {
                //the source is not NULL, and the code works correctly. This can be checked by adding the return type in the query(type: __typename). Thus, the result of the query will contain the type.
                if (source is FailureResultSource)
                    return failureResultType;
                else
                    return successResultType;
            };
        }
    }

query:
mutation ExecuteStepCommand($input:TestCommandInput!){
  checkout{
    executeCommand(input:$input){
      ... on TestSuccessResult{
        result
      }
      ... on TestFailureResult{
        reason
      }
    }
  }
};

response:
{"data":{"checkout":{"executeCommand":{}}}}
```

### Environment
Version: 2.4

Could you check what is wrong? I saw your tests for this functionality, but there is nothing in the tests that checks the ResolveType in Unions.

**Comment by sungam3r** [comment]
v2.4.0 was published more than 2 years ago, v3.0.0 was published 4 months ago with **a lot** of bugfixes and now we are almost ready for publishing v4.0.0 so I recommend you to migrate. Most likely described error has already been fixed. But if you can do a PR and write a test that reproduces the problem, then great. But in any case, the fix will be applied to the current "working" versions - 4.0.0 in case of breaking changes and 3.x.x in case of backward compatible changes.

**Comment by sungam3r** [comment]
Closed due to inactivity.
